web/docs/general/wasp-ts-config.md

```diff
@@ -0,0 +1,243 @@
+
+## How to switch from Wasp DSL config to Wasp TS config
+
+1. Go into your Wasp project that you want to switch to Wasp TS config (or create a new Wasp project if you want to try it out like that). Make sure you are on Wasp >= 0.15 and your project is working.
+2. Rename `tsconfig.json` file to `tsconfig.src.json` and add `"include": ["src"]` entry to it:
+
+    ```json title="tsconfig.src.json"
+    {
+      "compilerOptions": {
+        ...
+      },
+      ...
+      "include": ["src"]
+    }
+    ```
+
+3. Create new `tsconfig.json` file with the following content:
+
+    ```json title="tsconfig.json"
+    {
+      "files": [],
+      "references": [
+        { "path": "./tsconfig.src.json" },
+        { "path": "./tsconfig.wasp.json" }
+      ]
+    }
+    ```
+
+4. Create new `tsconfig.wasp.json` file with the following content:
+
+    ```json title="tsconfig.wasp.json"
+    {
+      "compilerOptions": {
+        "skipLibCheck": true,
+        "target": "ES2022",
+        "isolatedModules": true,
+        "moduleDetection": "force",
+
+        // linting
+        "strict": true,
+        "noUnusedLocals": true,
+        "noUnusedParameters": true,
+
+        "module": "NodeNext",
+        "noEmit": true,
+
+        "lib": ["ES2023"],
+      },
+      "include": ["main.wasp.ts"]
+    }
+    ```
+
+5. In your `package.json`, add `"type": "module"` entry at the top level:
+
+    ```json title="package.json"
+    {
+      "type": "module",
+      ...
+    }
+    ```
+
+6. Rename `main.wasp` file to `main.wasp.old`: we want to keep it for now so we can look at it while writing `main.wasp.ts` based on it.
+7. Run `wasp clean` and `rm package-lock.json` to ensure we have a clean slate to start from.
+8. Run `wasp ts-setup` to do the basic setup for Wasp TS config. It will also add `wasp-config` package to `devDependencies` in your `package.json`.
+9. Create an empty `main.wasp.ts` file and rewrite your `main.wasp.old` in it but in TypeScript.
+
+   Check out the [reference main.wasp.ts file](#reference-main-wasp-ts-file) below for details on what the TypeScript API for configuring Wasp looks like.
+In short, you will want to import the `App` from `wasp-config`, use it to define parts of your web app like `auth`, `pages`, `query`, `api`, ..., and at the end do the `default export app`.
+
+   You can manually do the rewrite using the reference file and TS types as guides (IDE support should work for you in `main.wasp.ts`), or you can (and we recommend it!) give the reference main.wasp.ts file to the LLM of your choice and tell it to rewrite your `main.wasp` while following the format in the reference file: we had great results with this!
+10. Run `wasp start` to run your app! If you got it all right, it should work the same as it worked before: the only difference is that it is now reading the Wasp config from `main.wasp.ts` instead of `main.wasp`.
```

Comment:
```suggestion
10. Run `wasp start` to run your app! If you got everything right, your app should work exactly like it did before. The only difference is that it's now reading the Wasp config from `main.wasp.ts` instead of `main.wasp`.
```